packages/marshal/src/helpers/promise.js

```diff
@@ -0,0 +1,45 @@
+// @ts-check
+
+/// <reference types="ses"/>
+
+import { isPromise } from '@endo/promise-kit';
+import { assertChecker, PASS_STYLE } from './passStyle-helpers.js';
+import { assertSafePromise } from './safe-promise.js';
+
+const { details: X, quote: q } = assert;
+const { ownKeys } = Reflect;
+const { getPrototypeOf, prototype: objectPrototype } = Object;
+
+/**
+ *
+ * @type {import('./internal-types.js').PassStyleHelper}
+ */
+export const PromiseHelper = harden({
+  styleName: 'promise',
+
+  canBeValid: (candidate, check) => {
+    return (
+      (isPromise(candidate) ||
+      candidate[PASS_STYLE] === 'promise' ||
+      check(false, X`Missing ${q(PASS_STYLE)}: ${candidate}`))
+    );
+  },
+
+  assertValid: candidate => {
+    if (isPromise(candidate)) {
+      assertSafePromise(candidate);
+      return;
+    }
+
+    PromiseHelper.canBeValid(candidate, assertChecker);
+
+    const proto = getPrototypeOf(candidate);
+    proto === objectPrototype ||
+      proto === null ||
+      assert.fail(X`Unexpected prototype for: ${candidate}`);
+
+    const descKeys = ownKeys(candidate);
+    (descKeys.length === 1 && descKeys[0] === PASS_STYLE) ||
+      assert.fail(X`Must not have any own properties: ${candidate}`);
```

Comment:
I thought that for @FUDCo's use case, we needed at least a `toString()` method? When discussing this with @erights we didn't see a need to disallow any more methods than for remotables.

Reply:
Being able to specify the `toString()` method is literally the only thing that matters to me here. Everything else is just machinery to enable that to work.

Reply:
> ... `toString()` method ...

Cannot do that in any case, and this PR wisely does not attempt. What do you really need? What are you doing now, so that your current state works? My sense is that you only need a pair of functions which can be local to a context, where

```
getSomething(makePromiseWithSomething(something)) === something
```
for some appropriate somethings. We can do that trivially with real promises and a weakmap, yes?

Reply:
> When discussing this with @erights we didn't see a need to disallow any more methods than for remotables.

True. And what I said then contradicts what I said above. There are two simple extremes, and we should do one or the other.

* Allow mostly-arbitrary unvalidatable methods, like we do for remotables. This is why remotables can have an arbitrary `toString` method.
* Data-only, which is where this PR currently is, and why I'm trying to avoid any unvalidatable behavior.

The problem is that we are still operating under the "any module might be multiply instantiated" restriction, and so any objective predicate, like `passStyleOf`, cannot brand things to remember that they are safe by construction. It can only tell that it is safe by inspection. `passStyleOf` does use a memo table to avoid redundant work, which is much like a brand table. Except that if absent, it always falls back to inspection. The behavior of non-primordial non-global methods cannot be inspected to be safe.

Reply:
This comes down to a tension between practicality and the principle that unmarshaling should recover full state. We can relax this constraint if the use case is in fact sufficiently strong, but we can't go in the other direction.